src/AbstractAlgebra.jl

```diff
@@ -1233,13 +1224,26 @@ export Generic
 
 ###############################################################################
 #
-#   Polynomial Ring S, x = R["x"] syntax
+#   misc
+#
+###############################################################################
+
+include("misc/ProductIterator.jl")
+include("misc/Evaluate.jl")
+include("misc/VarNames.jl")
+
+###############################################################################
+#
+#   Polynomial Ring S, x = R[:x] syntax
 #
 ###############################################################################
 
 getindex(R::NCRing, s::VarName) = polynomial_ring(R, s)
-getindex(R::NCRing, s::VarName, ss::VarName...) =
-   polynomial_ring(R, [Symbol(x) for x in (s, ss...)])
+# For compatibility `R[:x, :y]` returns `S, [x, y]` instead of `S, x, y`
+getindex(R::Ring, s::VarNames...) = ((S, x...) = polynomial_ring(R, s...); (S, collect(x)))
```

Comment:
Fine. I will add the dummy `VarNames` again.

Actually, I kept the 1 arg version. It is responsible for creating the univariate polynomial rings. I could remove it, if it were not for the compatibility `collect` usage.

I use `Ring` because `NCRing` does not allow for multivariate polynomial rings.